clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -5403,6 +5403,65 @@ def AtomicCmpXchg : CIR_Op<"atomic.cmp_xchg",
   let hasVerifier = 0;
 }
 
+def MemScope_SingleThread : I32EnumAttrCase<"MemScope_SingleThread",
+                                            0, "single_thread">;
+def MemScope_System       : I32EnumAttrCase<"MemScope_System",
+                                            1, "system">;
+
+def MemScopeKind : I32EnumAttr<
+    "MemScopeKind",
+    "Memory Scope Enumeration",
+    [MemScope_SingleThread, MemScope_System]> {
+  let cppNamespace = "::cir";
+}
+
+def AtomicOrdering_Relaxed
+  : I32EnumAttrCase<"AtomicOrdering_Relaxed",
+                    0, "relaxed">;
+def AtomicOrdering_Aquire
+  : I32EnumAttrCase<"AtomicOrdering_Aquire",
+                    1, "aquire">;
+def AtomicOrdering_Release
+  : I32EnumAttrCase<"AtomicOrdering_Release",
+                    2, "release">;
+def AtomicOrdering_AquireRelease
+  : I32EnumAttrCase<"AtomicOrdering_AquireRelease",
+                    3, "acq_rel">;
+def AtomicOrdering_SequentialConsistency
+  : I32EnumAttrCase<"AtomicOrdering_SequentialConsistency",
+                    4, "seq_cst">;
+
+def AtomicOrderingKind : I32EnumAttr<
+    "AtomicOrderingKind",
+    "Atomic Ordering Enumeration",
+    [AtomicOrdering_Relaxed, AtomicOrdering_Aquire,
+     AtomicOrdering_AquireRelease, AtomicOrdering_SequentialConsistency]> {
+  let cppNamespace = "::cir";
+}
+
+def AtomicFenceOp : CIR_Op<"atomic.thread_fence"> {
+  let summary = "Atomic thread fence";
+  let description = [{
+    C/C++ Atomic thread fence synchronization primitive. Implements the builtin
+    `__atomic_thread_fence` which enforces memory ordering constraints across
```

Comment:
If my reading of the OG code is right, this is going to be used by both {thread,signal}_fence, so `atomic.fence` is probably fine as a name. You should also mention in the description that this handling all variations of `*{thread,signal}_fence`